### PR TITLE
Cleans up and adds more concise admin logging to the centcom podlauncher

### LINKED ISSUE #40688: Minor supply pod smite issues
1) Canceling smite -> supply pod (quick) with the cancel button gives a popup: `ERROR: NULL path given.`
2) Smite -> supply pod logs `ADMIN LOG: Ckey/(George Melons) punished *no key*/(Pete) with Supply Pod.` as soon as you click it - even if you close the window and don't actually smash someone with the pod.

### PR DESCRIPTION
**Pull request by MrDoomBringer**
Fixes: #40688

:cl: MrDoomBringer
admin: The centcom podlauncher now has better logging
fix: Cancelling a supplypod smite will now not throw an error message
/:cl:

### CODE REVIEW

**Comment by Qustinnus** [comment]
you just want a t-shirt too

**Comment by MrDoomBringer** [comment]
yes

**Comment by ShizCalev** [comment]
conflict

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/cargo/centcom_podlauncher.dm [hunk 1]
```diff
@@ -403,12 +403,17 @@ force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.adm
...
+				target = get_turf(specificTarget) //if we have a specific target, then always launch the pod at the turf of the target
 			else if (target)
 				target = get_turf(target) //Make sure we're aiming at a turf rather than an item or effect or something
 			else
 				return //if target is null and we don't have a specific target, cancel
-
+			var/mob/M = locate(/mob/living, target)
+			if (M)
+				for (var/mob/bouttadie in target)
+					supplypod_punish_log(bouttadie)
```
[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/cargo/centcom_podlauncher.dm]: @AnturK okie

[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/cargo/centcom_podlauncher.dm]: when i said 2 hours i meant 8 days ok

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/cargo/centcom_podlauncher.dm [hunk 2]
```diff
@@ -514,4 +517,26 @@ force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.adm
...
\ No newline at end of file
+	. = ..()
+
+/datum/centcom_podlauncher/proc/supplypod_punish_log(var/list/whoDyin)
+	var/podString = effectBurst ? "5 pods" : "a pod"
+	var/whomString = ""
+	if (!isemptylist(whoDyin))
+		whomString = "at "
+		for (var/mob/living/M in whoDyin)
+			whomString += "[M], "
```
[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/cargo/centcom_podlauncher.dm]: o right

[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/cargo/centcom_podlauncher.dm]: ill do key_name cus thats what we use in randomverb punish logging

File: code/modules/cargo/centcom_podlauncher.dm [hunk 3]
```diff
@@ -514,4 +517,26 @@ force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.adm
...
 	qdel(temp_pod) //Delete the temp_pod
 	qdel(selector) //Delete the selector effect
-	. = ..()
\ No newline at end of file
+	. = ..()
+
+/datum/centcom_podlauncher/proc/supplypod_punish_log(var/list/whoDyin)
+	var/podString = effectBurst ? "5 pods" : "a pod"
+	var/whomString = ""
+	if (!isemptylist(whoDyin))
```
Comment (AnturK) [inline code/modules/cargo/centcom_podlauncher.dm]: Mind dropping this helper and just doing direct list check + null check / lazylen

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by AnturK** [comment]
Oh, didn't notice there was one more of these below.

**Pull request merged**